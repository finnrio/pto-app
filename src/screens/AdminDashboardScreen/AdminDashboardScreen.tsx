import React from "react";
import { Text, TouchableOpacity, View } from "react-native";
import styles from "./styles";

export default function AdminDashboardScreen({ navigation }: any) {
  return (
    <View style={styles.container}>
      <Text style={styles.text}>
        Welcome to the Admin dashboard. Here you can register and manage users.
      </Text>
      <TouchableOpacity
        style={styles.button}
        onPress={() => navigation.navigate("Admin User Management")}
        testID="user_management_btn"
      >
        <Text style={styles.buttonTitle}>Manage Users</Text>
      </TouchableOpacity>

      <TouchableOpacity
        style={styles.button}
        onPress={() => navigation.navigate("Register a User")}
        testID="register_user_btn"
      >
        <Text style={styles.buttonTitle}>Register a User</Text>
      </TouchableOpacity>
    </View>
  );
}
