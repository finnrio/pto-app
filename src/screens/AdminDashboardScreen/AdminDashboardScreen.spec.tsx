import {
  cleanup,
  fireEvent,
  render,
  screen,
} from "@testing-library/react-native";
import React from "react";
import AdminDashboardScreen from "./AdminDashboardScreen";

const navigation = {
  navigate: jest.fn(),
};

describe("AdminDashboardScreen", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });
  it("matches the snapshot", () => {
    const tree = render(<AdminDashboardScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  describe("User Management Button", () => {
    it("takes the user to the 'User Management' screen", () => {
      render(<AdminDashboardScreen navigation={navigation} />);
      fireEvent.press(screen.getByTestId("user_management_btn"));
      expect(navigation.navigate).toHaveBeenCalledWith("Admin User Management");
    });
  });
  describe("Register a User Button", () => {
    it("takes the user to the 'Register a User' screen", () => {
      render(<AdminDashboardScreen navigation={navigation} />);
      fireEvent.press(screen.getByTestId("register_user_btn"));
      expect(navigation.navigate).toHaveBeenCalledWith("Register a User");
    });
  });
});
