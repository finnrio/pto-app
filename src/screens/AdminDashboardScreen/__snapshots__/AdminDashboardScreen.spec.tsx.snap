// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminDashboardScreen matches the snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <Text
    style={
      {
        "marginBottom": 10,
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 10,
        "paddingLeft": 16,
      }
    }
  >
    Welcome to the Admin dashboard. Here you can register and manage users.
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#788eec",
        "borderRadius": 5,
        "height": 48,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 20,
        "opacity": 1,
      }
    }
    testID="user_management_btn"
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 18,
          "fontWeight": "bold",
          "padding": 10,
        }
      }
    >
      Manage Users
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#788eec",
        "borderRadius": 5,
        "height": 48,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 20,
        "opacity": 1,
      }
    }
    testID="register_user_btn"
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 18,
          "fontWeight": "bold",
          "padding": 10,
        }
      }
    >
      Register a User
    </Text>
  </View>
</View>
`;
